src/iop/indexer.rs

```diff
@@ -537,11 +521,12 @@ fn balance_matrices<F: Field>(
 
 pub(crate) fn post_process_matrices<F: PrimeField>(
     cs: &mut ConstraintSystem<F>,
+    domain_h: &Box<dyn EvaluationDomain<F>>,
+    domain_x: &Box<dyn EvaluationDomain<F>>,
 ) -> Option<(SparseMatrix<F>, SparseMatrix<F>, SparseMatrix<F>)> {
-    balance_matrices(&mut cs.at, &mut cs.bt);
```

Comment:
Now function `to_matrix_helper()`needs to take in input `domain_h` and `domain_x`, which are returned by function `build_domains()`. And `build_domains()` should be called after having called `balance_matrices()`, because balancing the matrices could possibly affect the size of the domains (more precisely of `domain_k` and `domain_b`).